src/mscorlib/src/System/Globalization/TextInfo.Windows.cs

```diff
@@ -49,44 +41,44 @@ private unsafe string ChangeCase(string s, bool toUpper)
             {
                 return s;
             }
-            else
+
+            int ret;
+
+            // Check for Invariant to avoid A/V in LCMapStringEx
+            uint linguisticCasing = IsInvariantLocale(_textInfoName) ? 0 : LCMAP_LINGUISTIC_CASING;
+
+            //
+            //  Create the result string.
+            //
+            string result = string.FastAllocateString(nLengthInput);
+
+            fixed (char* pSource = s)
+            fixed (char* pResult = result)
+            {
+                ret = Interop.Kernel32.LCMapStringEx(_sortHandle != IntPtr.Zero ? null : _textInfoName,
+                                                    toUpper ? LCMAP_UPPERCASE | linguisticCasing : LCMAP_LOWERCASE | linguisticCasing,
```

Comment:
NIT:  Would this be clearer as `(lingusticCasing | (toupper ? LCMAP_UPPERCASE : LCMAP_LOWERCASE))`.

Your call.